java-extensions/java-proxy/src/main/java/com/starrocks/javaproxy/HdfsRpcHandler.java

```diff
@@ -0,0 +1,230 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.javaproxy;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalListener;
+import com.google.common.cache.RemovalNotification;
+import com.google.protobuf.ByteString;
+import io.grpc.stub.StreamObserver;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.DFSInputStream;
+import org.apache.hadoop.hdfs.client.HdfsDataInputStream;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class HdfsRpcHandler extends PBackendServiceGrpc.PBackendServiceImplBase {
+    private final Logger logger = LogManager.getLogger(HdfsRpcHandler.class.getCanonicalName());
+    private final AtomicLong rpcOnFly = new AtomicLong();
+
+    private static class CacheValue {
+        public FileSystem fs;
+        public Path path;
+        public FSDataInputStream inputStream;
+        public ByteBuffer buffer;
+    }
+
+    private Configuration configuration;
+    private Cache<String, CacheValue> cache;
+
+    public HdfsRpcHandler() {
+        Configuration conf = new Configuration();
+        conf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+        conf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+        configuration = conf;
+
+        cache = CacheBuilder.newBuilder()
+                .maximumSize(128 * 1024)
```

Comment:
suggest make these parameter configurable

Reply:
Sure. Right now it's just a prototye.